partner_credit_limit/model/account_invoice.py

```diff
@@ -9,10 +9,16 @@ class AccontInvoice(models.Model):
 
     _inherit = 'account.invoice'
 
+    @api.multi
+    def check_payment_type(self):
+        if self.filtered(lambda invoice:
+                         invoice.payment_term_id.payment_type != 'credit'):
+            return True
+
     @api.multi
     def check_limit_credit(self):
         for invoice in self:
-            if invoice.payment_term_id.payment_type != 'credit':
+            if invoice.check_payment_type():
```

Comment:
is posible filtered into for?

Reply:
I don't understand how can I use filtered here.

Reply:
`self.filtered(lambda inv: inv.check_payment_type()):`

Reply:
Roger that. Change is [here](https://github.com/Vauxoo/addons-vauxoo/pull/1260/commits/4e6c2e3893fbf50e90b2f1bdf5646d1e97ff9f57#diff-b0121e38e8533ca630f1b3d6fc3e8d2aR19)